File: src/Gui/tasks/routes/api_email_verifications.py
"""
********************************************************************************************

Url Prefix: /api/email-verifications

********************************************************************************************
"""

from __future__ import annotations
import flask
from tasks.common import security
from tasks import services
from http import HTTPStatus

# module blueprint
bp_api_email_verifications = flask.Blueprint('api_email_verifications', __name__)

#------------------------------------------------------
# POST: /api/email-verifications
#------------------------------------------------------
@bp_api_email_verifications.post('')
@security.login_required
def post():
    response = services.email_verifications.send_email_verification()
    return (response.text, response.status_code)

File: src/Gui/tasks/static/js/custom/api/api-email-verifications.js

import { ApiEndpoints, HttpMethods } from "./api-base";

export class ApiEmailVerifications
{
    /**
     * Send post request
     * @returns {Promise<Response>}
     */
    post = async () =>
    {
        const url = ApiEndpoints.EMAIL_VERIFICATIONS;

        return await fetch(url, {
            method: HttpMethods.POST,
        })
    }
}

